src/lib/pubkey/hss_lms/lm_ots.cpp

```diff
@@ -0,0 +1,357 @@
+/**
+ * LM-OTS - Leighton-Micali One-Time Signatures
+ * (C) 2023 Jack Lloyd
+ *     2023 Fabian Albert, Philippe Lieser - Rohde & Schwarz Cybersecurity GmbH
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#include <botan/internal/lm_ots.h>
+
+#include <botan/exceptn.h>
+#include <botan/strong_type.h>
+#include <botan/internal/bit_ops.h>
+#include <botan/internal/hss_lms_utils.h>
+#include <botan/internal/safeint.h>
+
+namespace Botan {
+
+namespace {
+constexpr uint16_t D_PBLC = 0x8080;
+constexpr uint16_t D_MESG = 0x8181;
+/// For derivation of C as in https://github.com/cisco/hash-sigs
+constexpr uint16_t C_INDEX = 0xFFFD;
+
+class Chain_Generator {
+   public:
+      Chain_Generator(const LMS_Identifier& identifier, LMS_Tree_Node_Idx q) : m_gen(identifier) {
+         m_gen.set_q(q.get());
+      }
+
+      void process(HashFunction& hash,
+                   uint16_t chain_idx,
+                   uint8_t start,
+                   uint8_t end,
+                   std::span<const uint8_t> in,
+                   std::span<uint8_t> out) {
+         BOTAN_ARG_CHECK(start <= end, "Start value is bigger than end value");
+
+         if(start == end) {
+            copy_into(out, in);
+            return;
+         }
+         m_gen.set_i(chain_idx);
+
+         // Unroll first iteration of the loop
+         m_gen.set_j(start++);
+         m_gen.gen(out, hash, in);
```

Comment:
Yeah, since this is only one extra copy per hash chain, this is ok, I guess. It improves readability a lot 👍 
I can also remove `if(start == end) { return; }` since it is covered by the loop.